rvm install rbx fails on FreeBSD 8.0

On a FreeBSD 8.0 machine, I installed rvm with:

```
bash < <(curl -s https://rvm.beginrescueend.com/install/rvm)
```

Then I loaded it with `source ~/.rvm/scripts/rvm` and tried to run:

```
rvm install rbx
```

Which failed during the build process with:

<pre>1: LD vm/vm
vm/artifacts/environment.cpp.o(.text+0x2bd): In function `rubinius::Environment::start_signals()':
/usr/include/c++/4.2/ext/new_allocator.h:107: undefined reference to `pthread_create'
vm/artifacts/environment.cpp.o(.text+0x587): In function `rubinius::Environment::start_agent(int)':
vm/configuration.hpp:107: undefined reference to `pthread_create'
vm/artifacts/signal.cpp.o(.text+0x467): In function `rubinius::SignalHandler::on_fork()':
vm/util/thread.hpp:122: undefined reference to `pthread_create'
vm/builtin/artifacts/system.cpp.o(.text+0x10e9): In function `rubinius::System::vm_agent_io(rubinius::VM*)':
vm/util/thread.hpp:122: undefined reference to `pthread_create'vm/builtin/artifacts/thread.cpp.o(.text+0x110): In function `rubinius::Thread::fork(rubinius::VM*)':vm/util/thread.hpp:122: undefined reference to `pthread_create'vm/builtin/artifacts/thread.cpp.o(.text+0x253): In function `rubinius::Thread::priority(rubinius::VM*)':vm/util/thread.hpp:169: undefined reference to `pthread_getschedparam'vm/builtin/artifacts/thread.cpp.o(.text+0x703): In function `rubinius::Thread::set_priority(rubinius::VM*, rubinius::Fixnum*)':vm/util/thread.hpp:178: undefined reference to `pthread_getschedparam'vm/builtin/artifacts/thread.cpp.o(.text+0x731):vm/util/thread.hpp:197: undefined reference to `pthread_setschedparam'vm/builtin/artifacts/thread.cpp.o(.gnu.linkonce.t._ZN6thread6Thread8priorityEv+0x15): In function `thread::Thread::priority()':vm/gc/root.hpp:108: undefined reference to `pthread_getschedparam'vm/builtin/artifacts/thread.cpp.o(.gnu.linkonce.t._ZN6thread6Thread12set_priorityEi+0x2e): In function `thread::Thread::set_priority(int)':vm/util/thread.hpp:128: undefined reference to `pthread_getschedparam'vm/builtin/artifacts/thread.cpp.o(.gnu.linkonce.t._ZN6thread6Thread12set_priorityEi+0x64):vm/util/thread.hpp:197: undefined reference to `pthread_setschedparam'Error: gcc -o vm/vm vm/artifacts/agent.cpp.o vm/artifacts/agent_components.cpp.o vm/artifacts/arguments.cpp.o vm/artifacts/bytecode_verification.cpp.o vm/artifacts/call_frame.cpp.o vm/artifacts/compiled_file.cpp.o vm/artifacts/config_parser.cpp.o vm/artifacts/dispatch.cpp.o vm/artifacts/environment.cpp.o vm/artifacts/exception.cpp.o vm/artifacts/exception_point.cpp.o vm/artifacts/ffi.cpp.o vm/artifacts/ffi_util.cpp.o vm/artifacts/global_cache.cpp.o vm/artifacts/global_lock.cpp.o vm/artifacts/helpers.cpp.o vm/artifacts/inline_cache.cpp.o vm/artifacts/instructions.cpp.o vm/artifacts/kcode.cpp.o vm/artifacts/linkedlist.cpp.o vm/artifacts/marshal.cpp.o vm/artifacts/native_thread.cpp.o vm/artifacts/objectmemory.cpp.o vm/artifacts/ontology.cpp.o vm/artifacts/oop.cpp.o vm/artifacts/primitives.cpp.o vm/artifacts/shared_state.cpp.o vm/artifacts/signal.cpp.o vm/artifacts/stack_variables.cpp.o vm/artifacts/symboltable.cpp.o vm/artifacts/thread_state.cpp.o vm/artifacts/type_info.cpp.o vm/artifacts/virtual.cpp.o vm/artifacts/vm.cpp.o vm/artifacts/vmmethod.cpp.o vm/builtin/artifacts/access_variable.cpp.o vm/builtin/artifacts/alias.cpp.o vm/builtin/artifacts/array.cpp.o vm/builtin/artifacts/autoload.cpp.o vm/builtin/artifacts/bignum.cpp.o vm/builtin/artifacts/block_as_method.cpp.o vm/builtin/artifacts/block_environment.cpp.o vm/builtin/artifacts/bytearray.cpp.o vm/builtin/artifacts/call_unit.cpp.o vm/builtin/artifacts/call_unit_adapter.cpp.o vm/builtin/artifacts/capi_handle.cpp.o vm/builtin/artifacts/channel.cpp.o vm/builtin/artifacts/class.cpp.o vm/builtin/artifacts/compactlookuptable.cpp.o vm/builtin/artifacts/compiledmethod.cpp.o vm/builtin/artifacts/data.cpp.o vm/builtin/artifacts/dir.cpp.o vm/builtin/artifacts/exception.cpp.o vm/builtin/artifacts/executable.cpp.o vm/builtin/artifacts/ffi_pointer.cpp.o vm/builtin/artifacts/fiber.cpp.o vm/builtin/artifacts/find_object.cpp.o vm/builtin/artifacts/fixnum.cpp.o vm/builtin/artifacts/float.cpp.o vm/builtin/artifacts/global_cache_entry.cpp.o vm/builtin/artifacts/heap_dump.cpp.o vm/builtin/artifacts/immediates.cpp.o vm/builtin/artifacts/integer.cpp.o vm/builtin/artifacts/io.cpp.o vm/builtin/artifacts/iseq.cpp.o vm/builtin/artifacts/list.cpp.o vm/builtin/artifacts/location.cpp.o vm/builtin/artifacts/lookuptable.cpp.o vm/builtin/artifacts/methodtable.cpp.o vm/builtin/artifacts/module.cpp.o vm/builtin/artifacts/nativefunction.cpp.o vm/builtin/artifacts/nativemethod.cpp.o vm/builtin/artifacts/object.cpp.o vm/builtin/artifacts/pack.cpp.o vm/builtin/artifacts/packed_object.cpp.o vm/builtin/artifacts/proc.cpp.o vm/builtin/artifacts/randomizer.cpp.o vm/builtin/artifacts/regexp.cpp.o vm/builtin/artifacts/staticscope.cpp.o vm/builtin/artifacts/string.cpp.o vm/builtin/artifacts/symbol.cpp.o vm/builtin/artifacts/system.cpp.o vm/builtin/artifacts/thread.cpp.o vm/builtin/artifacts/thunk.cpp.o vm/builtin/artifacts/time.cpp.o vm/builtin/artifacts/tuple.cpp.o vm/builtin/artifacts/unpack.cpp.o vm/builtin/artifacts/variable_scope.cpp.o vm/builtin/artifacts/weakref.cpp.o vm/capi/artifacts/array.cpp.o vm/capi/artifacts/bignum.cpp.o vm/capi/artifacts/capi.cpp.o vm/capi/artifacts/class.cpp.o vm/capi/artifacts/data.cpp.o vm/capi/artifacts/exception.cpp.o vm/capi/artifacts/float.cpp.o vm/capi/artifacts/gc.cpp.o vm/capi/artifacts/globals.cpp.o vm/capi/artifacts/handle.cpp.o vm/capi/artifacts/hash.cpp.o vm/capi/artifacts/io.cpp.o vm/capi/artifacts/kernel.cpp.o vm/capi/artifacts/marshal.cpp.o vm/capi/artifacts/module.cpp.o vm/capi/artifacts/numeric.cpp.o vm/capi/artifacts/object.cpp.o vm/capi/artifacts/proc.cpp.o vm/capi/artifacts/range.cpp.o vm/capi/artifacts/regexp.cpp.o vm/capi/artifacts/string.cpp.o vm/capi/artifacts/struct.cpp.o vm/capi/artifacts/symbol.cpp.o vm/capi/artifacts/thread.cpp.o vm/capi/artifacts/time.cpp.o vm/capi/artifacts/util.cpp.o vm/drivers/artifacts/cli.cpp.o vm/gc/artifacts/baker.cpp.o vm/gc/artifacts/code_manager.cpp.o vm/gc/artifacts/debug.cpp.o vm/gc/artifacts/gc.cpp.o vm/gc/artifacts/heap.cpp.o vm/gc/artifacts/immix.cpp.o vm/gc/artifacts/inflated_headers.cpp.o vm/gc/artifacts/managed.cpp.o rake aborted!</pre>


After analyzing the failing link command, I realized that `-pthread` isn't being added to the `LDFLAGS`, so I tried to figure out where that was being set and eventually landed inside the `rbx-head/rakelib/blueprint.rb` file. When evaluating `case RUBY_PLATFORM`, there's a case for openbsd that sets `-pthread`, but the normal bsd case (which I believe I was hitting) only sets `-lcrypt -rdynamic` and not `-pthread`. I added the following under `case RUBY_PLATFORM`:

```
when /freebsd/i
  gcc.ldflags << '-lcrypt' << '-pthread' << '-lssl' << "-rdynamic" << "-Wl,--export-dynamic"
  make = "gmake"
```

And now it builds successfully when I run `rake install`.

--- a/rakelib/blueprint.rb
+++ b/rakelib/blueprint.rb
@@ -44,6 +44,9 @@ Daedalus.blueprint do |i|
   case RUBY_PLATFORM
   when /linux/i
     gcc.ldflags << '-Wl,--export-dynamic' << "-lrt" << "-lcrypt" << "-ldl" << "-lpthread"
+  when /freebsd/i
+    gcc.ldflags << '-lcrypt' << '-pthread' << '-rdynamic'
+    make = "gmake"
   when /openbsd/i
     gcc.ldflags << '-lcrypto' << '-pthread' << '-lssl' << "-rdynamic" << "-Wl,--export-dynamic"
     make = "gmake"

